runtime/src/qio/qio_formatted.c

```diff
@@ -171,20 +171,22 @@ qioerr _peek_until_byte(qio_channel_t* restrict ch, uint8_t term_byte, int64_t*
 static
 qioerr _peek_until_len(qio_channel_t* restrict ch, ssize_t len, int64_t* restrict amt_read_out)
 {
+
+#define BUFSIZE (16 * 1024)
   qioerr err;
   int64_t mark_offset = 0;
   int64_t end_offset = 0;
   uint64_t num = 0;
-  uint8_t byte = 0;
+  uint8_t buf[BUFSIZE];
   ssize_t count;
 
   mark_offset = qio_channel_offset_unlocked(ch);
 
   err = qio_channel_mark(false, ch);
   if( err ) return err;
 
-  for( count = 0; count < len; count++ ) {
-    err = qio_channel_read_uint8(false, ch, &byte);
+  for( count = 0; count < len; count += BUFSIZE) {
```

Comment:
I don't think this handles peeking by reading until something that isn't a multiple of BUFSIZE correctly. Isn't it just going to read to the smallest multiple of BUFSIZE and then leave the rest not read?

Reply:
Regarding your first suggestion, I'm inclined to leave it as-is because it shouldn't have much of an impact and I don't fully understand the underlying chunking functionality. I can always figure out how to make the changes if it turns out to be necessary.

Regarding peeking, I think it works correctly. `qio_channel_read_amt` returns an error on a short read, but does read the bytes and update the offset.